draft-ietf-quic-transport.md

```diff
@@ -2187,18 +2187,19 @@ of three ways:
 
 The closing and draining connection states exist to ensure that connections
 close cleanly and that delayed or reordered packets are properly discarded.
-These states SHOULD persist for three times the current Probe Timeout (PTO)
-interval as defined in {{QUIC-RECOVERY}}.
+These states SHOULD persist for at least three times the current Probe Timeout
+(PTO) interval as defined in {{QUIC-RECOVERY}}.
 
 An endpoint enters a closing period after initiating an immediate close
 ({{immediate-close}}).  While closing, an endpoint MUST NOT send packets unless
 they contain a CONNECTION_CLOSE frame (see {{immediate-close}} for details).  An
 endpoint retains only enough information to generate a packet containing a
 CONNECTION_CLOSE frame and to identify packets as belonging to the connection.
-The connection ID and QUIC version is sufficient information to identify packets
-for a closing connection; an endpoint can discard all other connection state.
-An endpoint MAY retain packet protection keys for incoming packets to allow it
-to read and process a CONNECTION_CLOSE frame.
+The source and the destination connection ID and the QUIC version are sufficient
```

Comment:
If it's possible the source CID could change, I'd suggest we only mention the destination CID.

Reply:
It looks like we have different scenarios in mind. I was thinking about an error that occurs during the handshake. In that case, you close the connection, but there might still be Handshake packets arriving, which you would want to ignore then.
You seem to be thinking about the peer switching to a new connection ID. If we want to cover that case, it seems like we need to remember *all* connection IDs that are still active.

Reply:
The intent here was destination connection ID.  That should always be sufficient to identify the connection.  The source connection ID doesn't appear on any packets that are likely to matter and it only matters for the purposes of generating these packets.

Source connection ID might matter for the first Initial packet from the client, where there might be some value in distinguishing between connection attempts with colliding destination connection IDs.  The odds of that are slim enough for this to not matter though.

Reply:
We're talking from the view point of a connection here, so the CID that's necessary to assign a packet is one of the *source* CIDs (which will be the *destination* CID of the incoming packet).
You'll need to remember at least one of the *destination* CIDs for the connection ID in order to repeat the CONNECTION_CLOSE frame in the closing state.

Reply:
I don't know what people view as source and destination when not talking about packets.  That's not something we have crisp terms for, but using source and destination here makes that problem worse.

Reply:
Just talking about "connection ID" is insufficient though, since I need to remember at least the connection ID I'm receiving packets at (which I would have called my source), and one connection ID I'm sending packets to (which I would have called my destination).
In addition, depending on how frequently the peer switches to new connection IDs, I maybe should also remember the outstanding connection IDs I've issued to my peer.

Reply:
This text is specifically talking about identifying incoming packets, so the only connection ID that is relevant is the inbound one.  The final sentence of the paragraph is where the outbound connection ID becomes relevant.